### ISSUE TITLE
Directus fails to connect to mysql

### ISSUE THREAD

**Issue by jsamr** [directus/directus#2197]
## Version Info
  - Directus version and branch (Or commit hash): 6.4.9
  - PHP version: 7.2
  - MySQL version: mysql  Ver 14.14 Distrib 5.5.60, for debian-linux-gnu (x86_64) using readline 6.3
  - Web server:  Apache
  - OS name and version: Debian 3.16.57-1~ui80+1 (2018-08-13) x86_64 GNU/Linux

## Expected Behavior

When I attempt login with good credentials, I should be redirected to my home dashboard.

## Actual Behavior

When I attempt login with good credentials, I receive a 403, permission denied error.

## Steps to Reproduce

This issue is specific to my client host (1&1 shared server).
I have tested mysql connection through SSH, with exactly the same credentials as in `config.php`, successfully.
The database has been populated with a snapshot which has been proven working in another host (a2hosting).

Bellow is a list of tables in this database:

``` txt
mysql> show tables; 
+------------------------------+
| Tables_in_db739409183        |
+------------------------------+
| accompaniment                |
| directus_activity            |
| directus_bookmarks           |
| directus_columns             |
| directus_files               |
| directus_groups              |
| directus_messages            |
| directus_messages_recipients |
| directus_preferences         |
| directus_privileges          |
| directus_schema_migrations   |
| directus_settings            |
| directus_tables              |
| directus_users               |
| info-contact                 |
| page-about                   |
| page-about-rewards           |
| page-accompaniment           |
| page-accompaniments          |
| page-contact                 |
| page-home                    |
| page-home-partners           |
| page-legal-notice            |
| partner                      |
| seo-info                     |
+------------------------------+
25 rows in set (0.01 sec)

```

## Schema Dump, Logs, or Screenshots
403 response:
```json
{
  "success":false,
  "error":{
    "message":"Permission denied"
  },
  "code":0,
  "class":"Directus\\Exception\\Http\\ForbiddenException",
  "file":"\/homepages\/32\/d367781095\/htdocs\/www\/admin\/api\/core\/Directus\/Database\/TableSchema.php",
  "line":183
}
```
Log: 
``` txt
ERROR - 2018-08-27T21:56:15-04:00 - exception 'Directus\Exception\Http\ForbiddenException' with message 'Permission denied' in /homepages/32/d367781095/htdocs/www/admin/api/core/Directus/Database/TableSchema.php:183
Stack trace:
#0 /homepages/32/d367781095/htdocs/www/admin/api/core/Directus/Database/TableGateway/BaseTableGateway.php(186): Directus\Database\TableSchema::getTableSchema('directus_files', Array, false, false)
#1 /homepages/32/d367781095/htdocs/www/admin/api/core/Directus/Database/TableGateway/BaseTableGateway.php(106): Directus\Database\TableGateway\BaseTableGateway->getTableSchema()
#2 /homepages/32/d367781095/htdocs/www/admin/api/core/Directus/Database/TableGatewayFactory.php(69): Directus\Database\TableGateway\BaseTableGateway->__construct('directus_files', Object(Directus\Database\Connection), Object(Directus\Permissions\Acl))
#3 /homepages/32/d367781095/htdocs/www/admin/api/core/functions.php(1387): Directus\Database\TableGatewayFactory::create('directus_files', Array)
#4 /homepages/32/d367781095/htdocs/www/admin/api/api.php(520): get_project_info()
#5 [internal function]: {closure}()
#6 /homepages/32/d367781095/htdocs/www/admin/vendor/slim/slim/Slim/Slim.php(598): call_user_func(Object(Closure))
#7 /homepages/32/d367781095/htdocs/www/admin/vendor/slim/slim/Slim/Slim.php(1367): Slim\Slim->notFound()
#8 /homepages/32/d367781095/htdocs/www/admin/vendor/slim/slim/Slim/Middleware/Flash.php(85): Slim\Slim->call()
#9 /homepages/32/d367781095/htdocs/www/admin/vendor/slim/slim/Slim/Middleware/MethodOverride.php(92): Slim\Middleware\Flash->call()
#10 /homepages/32/d367781095/htdocs/www/admin/api/core/Directus/Slim/CorsMiddleware.php(19): Slim\Middleware\MethodOverride->call()
#11 /homepages/32/d367781095/htdocs/www/admin/api/core/Directus/Slim/HttpCacheMiddleware.php(18): Directus\Slim\CorsMiddleware->call()
#12 /homepages/32/d367781095/htdocs/www/admin/api/core/Directus/Slim/ResponseCacheMiddleware.php(37): Directus\Slim\HttpCacheMiddleware->call()
#13 /homepages/32/d367781095/htdocs/www/admin/vendor/slim/slim/Slim/Slim.php(1302): Directus\Slim\ResponseCacheMiddleware->call()
#14 /homepages/32/d367781095/htdocs/www/admin/api/core/Directus/Application/Application.php(103): Slim\Slim->run()
#15 /homepages/32/d367781095/htdocs/www/admin/api/api.php(533): Directus\Application\Application->run()
#16 {main}
```
Screen:

![directusbug](https://user-images.githubusercontent.com/3646758/44697029-08fd8600-aa48-11e8-98ae-d1cca5e0b1f5.png)

**Comment by jsamr** [comment]
I would like to add that I could successfully add a user in database with directus CLI, which clearly means it's not related to database configuration:
```
php7.1 bin/directus user:password -e test@test.com -p test
```
```
mysql> select email from directus_users;
+------------------------------+
| email                        |
+------------------------------+
| xxxxx@xxxxx.xxxxx            |
| xxxxx.@xxxxx.xxxxx           |
| xxxxx.@xxxxx.xxxxx           |
| test@test.com                |
+------------------------------+
4 rows in set (0.00 sec)
```

**Comment by rijkvanzanten** [comment]
Could you check what happens when you install fresh on this server? It looks like the user you're trying to log in with doesn't have permission to read from the `directus_files` table, but I'm not 100% sure. Could you double check (in the DB directly) what group the user has you're trying to login with and if that user group has the right permissions to read the `directus_files` table?

**Comment by jsamr** [comment]
@rijkvanzanten Thank you very much for your help. Yes they do seem to have permissions : 

``` sql
select table_name,allow_view,group_id from directus_privileges where table_name='directus_files';
```
```
+----------------+------------+----------+
| table_name     | allow_view | group_id |
+----------------+------------+----------+
| directus_files |          2 |        1 |
| directus_files |          2 |        2 |
| directus_files |          2 |        3 |
+----------------+------------+----------+

All groups can read directus_files.

```
```sql
select id,`group` from directus_users;
```
```+----+-------+
| id | group |
+----+-------+
|  1 |     1 |
|  2 |     3 |
|  3 |     3 |
|  4 |     3 |
+----+-------+
```
One user is admin (the one I wanted to authenticate with, the others have write permission, see bellow select).

``` sql
select id,name from directus_groups;
```

```
+----+----------------------+
| id | name                 |
+----+----------------------+
|  1 | Administrator        |
|  3 | Éditeurs de contenu  |
|  2 | Public               |
+----+----------------------+
```

I will try what you suggested later in the day in maintenance time.

**Comment by jsamr** [comment]
@rijkvanzanten I figured it out. It was a routing issue, .htaccess related. Very close to #2144.

When I set directus path to `/admin` in `api/config.php`, I also have to modify (in filesystem) `/admin/.htaccess` and `/admin/api/.htaccess` with, respectively, the following rules : 

- `RewriteBase /admin`
- `RewriteBase /admin/api`

In that case, everything work as expected!

I suggest you add a section in the doc for situations where directus path `!= /`.

Thanks for your help, anyway!
